### PR TITLE
chore(Core): Debloat Mail.cpp

### PR DESCRIPTION
**Pull request by Synful-Syn**
## Changes Proposed:
- Debloat the following by process of elimination since `receiver` won't suddently change and avoid duplicate checks "are we missing `receiver`?".
image
image
image
- Don't do `ObjectAccessor::FindPlayerByLowGUID()` if we don't need it.
- ~~Don't calculate the value of `index` since it is always the same value for each `stmt->SetData(...)`.~~ (cancelled)
- Use directly the value of `mailItemIter->second` instead of putting it into an `Item*`. Same with `CharacterDatabaseTransaction(nullptr)`.

## Issues Addressed:

- Closes Nothing.

## Tests Performed:

- Got mails from player.
- Got mails from player with items
- Got mails from GM.
- Got mails from GM with items.
- Got mails from AH.
- Returned mails.
- Deleted mails.
- Checked the returned mails.

## How to Test the Changes:

Do mail stuff.

## Known Issues and TODO List:

- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Review by acidmanifesto — APPROVED** [review]

**Comment by acidmanifesto** [comment]
We need a Return to Sender Mail Test for this PR

File: src/server/game/Mails/Mail.cpp [hunk 1]
```diff
@@ -219,43 +212,41 @@ void MailDraft::SendMailTo(CharacterDatabaseTransaction trans, MailReceiver cons
...
-    stmt->SetData(++index, sender.GetSenderId());
-    stmt->SetData(++index, receiver.GetPlayerGUIDLow());
-    stmt->SetData(++index, GetSubject());
-    stmt->SetData(++index, GetBody());
-    stmt->SetData  (++index, !m_items.empty());
-    stmt->SetData(++index, uint32(expire_time));
-    stmt->SetData(++index, uint32(deliver_time));
-    stmt->SetData(++index, m_money);
-    stmt->SetData(++index, m_COD);
-    stmt->SetData (++index, uint8(checked));
```
Comment (Kitzunu) [inline src/server/game/Mails/Mail.cpp:250]: I prefer this as it is impossible to mess up if tables are added or removed.

Comment (Synful-Syn) [reply src/server/game/Mails/Mail.cpp:250]: I have "reverted" it. I think your point is absolutely valid.

**Comment by Gultask** [comment]
All of the provided above work. Also tested replying and returning mails and it works.

**Comment by Nefertumm** [comment]
Ty for your PR Syn!

**Pull request merged**